libcrux-ml-dsa/src/ml_dsa_65.rs

```diff
@@ -68,22 +68,33 @@ type SIMDUnit = crate::simd::avx2::AVX2SIMDUnit;
 #[cfg(not(feature = "simd256"))]
 type SIMDUnit = crate::simd::portable::PortableSIMDUnit;
 
+// For regular shake128 we only use portable.
+type Shake128 = crate::hash_functions::portable::Shake128;
+
+#[cfg(feature = "simd256")]
+type Shake128X4 = crate::hash_functions::simd256::Shake128;
+#[cfg(not(feature = "simd256"))]
+type Shake128X4 = crate::hash_functions::portable::Shake128X4;
+
 #[cfg(feature = "simd256")]
-type Shake128 = crate::hash_functions::portable::PortableShake128;
+type Shake256X4 = crate::hash_functions::simd256::Shake256X4;
 #[cfg(not(feature = "simd256"))]
-type Shake128 = crate::hash_functions::portable::PortableShake128;
+type Shake256X4 = crate::hash_functions::portable::Shake256X4;
 
+// TODO: This is all portable for now.
 #[cfg(feature = "simd256")]
-type Shake256 = crate::hash_functions::portable::PortableShake256;
+type Shake256 = crate::hash_functions::portable::Shake256;
 #[cfg(not(feature = "simd256"))]
-type Shake256 = crate::hash_functions::portable::PortableShake256;
+type Shake256 = crate::hash_functions::portable::Shake256;
 
 /// Generate an ML-DSA-65 Key Pair
 pub fn generate_key_pair(randomness: [u8; KEY_GENERATION_RANDOMNESS_SIZE]) -> MLDSA65KeyPair {
     let (signing_key, verification_key) = crate::ml_dsa_generic::generate_key_pair::<
         SIMDUnit,
         Shake128,
```

Comment:
I guess there is no nice way of packaging all of these Shake parameters into a single parameter?

Reply:
I don't think we want this in on e parameter. We can drop some of them again when everything is cleaned up. But there are different states and ideally we only pass through the types we actually need.